File: 🟡 Medium/2419 - Longest Subarray With Maximum Bitwise AND/_2419LongestSubarrayWithMaximumBitwiseAND.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int maxi = 0;
        int currLongestSeq = 0;
        int res = 0;

        for (int i = 0; i < nums.size(); i++) {
            if (maxi < nums[i]) {
                maxi = nums[i];
                currLongestSeq = 0;
                res = 0;
            }

            if (maxi == nums[i]) {
                currLongestSeq++;
            } else {
                currLongestSeq = 0;
            }

            res = max(res, currLongestSeq);
        }
        return res;
    }
};

File: 🟡 Medium/2419 - Longest Subarray With Maximum Bitwise AND/_2419LongestSubarrayWithMaximumBitwiseAND.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        maxi = 0
        currLongestSeq = 0
        res = 0

        for i in range(0, len(nums)):
            if maxi < nums[i]:
                maxi = nums[i]
                currLongestSeq = 0
                res = 0

            if maxi == nums[i]:
                currLongestSeq += 1
            else:
                currLongestSeq = 0

            res = max(res, currLongestSeq)

        return res
